### PR TITLE
Add benchmarks to show impact of ExemplarFilter

### PR DESCRIPTION
**Pull request by cijothomas**
Fixes #.

## Changes

Please provide a brief description of the changes here.

For significant contributions please make sure you have completed the following items:

* [ ] Appropriate `CHANGELOG.md` updated for non-trivial changes
* [ ] Design discussion issue #
* [ ] Changes in public API reviewed

### CODE REVIEW

File: test/Benchmarks/Metrics/ExemplarBenchmarks.cs [hunk 1]
```diff
@@ -112,5 +132,18 @@ public void HistogramWithTagReduction()
 
             this.histogramWithTagReduction.Record(random.Next(1000), tags);
         }
+
+        public class LowValueExemplarFilter : ExemplarFilter
```
Comment (cijothomas) [inline test/Benchmarks/Metrics/ExemplarBenchmarks.cs:136]: want to use this is a to-be-written doc, to show the importance of using Filter to improve perf. (And show that Filter is the best way to improve perf, as it short circuits a lot of code, as opposed to using custom Reservoirs)

Comment (utpilla) [inline test/Benchmarks/Metrics/ExemplarBenchmarks.cs:136]: Why not name this `HighValue` as well? It's a bit confusing to use "LowValueFilter" for `HighValueOnly`.

File: test/Benchmarks/Metrics/ExemplarBenchmarks.cs [hunk 2]
```diff
@@ -23,19 +23,21 @@
...
-|   HistogramNoTagReduction |          False | 256.5 ns | 4.84 ns | 4.53 ns |         - |
-| HistogramWithTagReduction |          False | 246.6 ns | 4.90 ns | 4.81 ns |         - |
-|   HistogramNoTagReduction |           True | 286.4 ns | 5.30 ns | 7.25 ns |         - |
-| HistogramWithTagReduction |           True | 293.6 ns | 5.77 ns | 7.09 ns |         - |
+|                    Method | ExemplarFilter |     Mean |   Error |  StdDev |
+|-------------------------- |--------------- |---------:|--------:|--------:|
+|   HistogramNoTagReduction |      AlwaysOff | 380.7 ns | 5.92 ns | 5.53 ns |
+| HistogramWithTagReduction |      AlwaysOff | 356.5 ns | 3.33 ns | 2.95 ns |
+|   HistogramNoTagReduction |       AlwaysOn | 412.3 ns | 2.11 ns | 1.64 ns |
+| HistogramWithTagReduction |       AlwaysOn | 461.0 ns | 4.65 ns | 4.35 ns |
```
Comment (utpilla) [inline test/Benchmarks/Metrics/ExemplarBenchmarks.cs:38]: Add another params for number of dimensions for benchmarks that do tag reduction?

Comment (cijothomas) [reply test/Benchmarks/Metrics/ExemplarBenchmarks.cs:38]: it'll clutter a lot.. as I am adding more and more to this now. Will find a balanced approach before finalizing.. Too much makes this cluttered. Too less makes it hard to convey which ones end-users should focus on..

**Review by utpilla — APPROVED** [review]

File: test/Benchmarks/Metrics/ExemplarBenchmarks.cs (hunk 2 above)
Comment (utpilla) [reply test/Benchmarks/Metrics/ExemplarBenchmarks.cs:38]: Yeah we could have them in a different benchmarks class.

**Pull request merged**